rudderc/src/ir/resource.rs

```diff
@@ -18,23 +19,43 @@ pub struct Policy {
 #[derive(Debug, PartialEq, Serialize, Deserialize)]
 pub struct Resource {
     pub name: String,
-    // named params?
-    pub params: Vec<String>,
+    pub params: HashMap<String, String>,
+    pub resource_type: String,
     // contains either states or resources
     pub states: Vec<State>,
+    pub id: String,
+    pub reporting: Option<ReportingPolicy>
 }
 
 #[derive(Debug, PartialEq, Serialize, Deserialize)]
 pub struct State {
-    pub name: String,
-    pub id: String,
     // TODO specific type with custom deserializer that check validity
     // class regex or variables
     pub condition: String,
+    pub id: String,
     pub meta: Value,
-    pub report_component: String,
+    pub name: String,
+    pub params: HashMap<String, String>,
     // comes from stdlib
     pub report_parameter: String,
-    // named params?
-    pub params: Vec<String>,
+    pub state_type: String,
 }
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct ReportingPolicy {
+    pub enabled: Option<bool>,
```

Comment:
I think these parameters shouldn't be `Option` but have a default value instead.

Reply:
With https://serde.rs/field-attrs.html#default--path or https://serde.rs/field-attrs.html#default

Reply:
:+1: